src/Payments.Mvc/Controllers/FinancialAccountsController.cs

```diff
@@ -156,17 +141,11 @@ public FinancialAccountsController(ApplicationDbContext context, IFinancialServi
             financialAccount.IsDefault = financialAccountModel.IsDefault;
             financialAccount.TeamId = financialAccountModel.TeamId;
 
-            var team = await _context.Teams
-                .SingleOrDefaultAsync(m => m.Id == financialAccount.TeamId && m.IsActive);
+            var team = await _context.Teams.SingleOrDefaultAsync(m => m.Id == financialAccount.TeamId && m.IsActive);

```

Comment:
You should keep using TeamSlug here. Otherwise they can submit this request for a different team.

Reply:
@jpknoll Ok, these and a couple others in this controller fixed.